src/server/serve.rs

```diff
@@ -435,6 +437,50 @@ fn parse_resolv_conf(content: &str) -> AardvarkResult<Vec<SocketAddr>> {
     Ok(nameservers)
 }
 
+async fn monitor_upstream_resolvers(nameservers: Arc<Mutex<Vec<SocketAddr>>>) {
+    let mut inotify = match Inotify::init() {
+        Ok(inotify) => inotify,
+        Err(err) => {
+            error!("Failed to initialize inotify: {err}");
+            warn!("Nameservers will not be updated on resolv.conf change");
+            return;
```

Comment:
Properly best to log this as one line.

Though from experience inotify can fail so I wonder if we should also implement a timer based polling fallback, i.e. read the resolv.conf every 5s or so.